imdclient/IMDClient.py

```diff
@@ -129,9 +139,11 @@ def stop(self):
         if self._multithreaded:
             if not self._stopped:
                 self._buf.notify_consumer_finished()
+                self._wait(self._wait_after_disconnect)
```

Comment:
A few things here

The IMDClient object has to be compatible with IMDv2 and IMDv3, so by calling this wait method directly in stop, you are potentially sending an IMDv3 packet to an IMDv2 receiver. The Producer object should handle the version specific details.

Maybe _wait could be called within the _disconnect method of the v3 producer>